common/src/main/java/com/wynntils/features/user/TradeMarketAutoOpenChatFeature.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.user;
+
+import com.wynntils.core.features.UserFeature;
+import com.wynntils.mc.event.ScreenClosedEvent;
+import com.wynntils.mc.utils.McUtils;
+import com.wynntils.wc.event.ChatMessageReceivedEvent;
+import com.wynntils.wc.utils.WynnUtils;
+import java.util.regex.Pattern;
+import net.minecraft.client.gui.screens.ChatScreen;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class TradeMarketAutoOpenChatFeature extends UserFeature {
+
+    // Type the price in emeralds or type 'cancel' to cancel:
+    // Type the amount you wish to buy or type 'cancel' to cancel:
+    // Type the item name or type 'cancel' to cancel:
+    private static final Pattern TYPE_TO_CHAT_PATTERN = Pattern.compile("§6Type the .* or type 'cancel' to cancel:");
+
+    private boolean openChatWhenContainerClosed = false;
+
+    @SubscribeEvent
+    public void onChatMessageReceive(ChatMessageReceivedEvent event) {
+        if (!WynnUtils.onWorld()) return;
+
+        if (TYPE_TO_CHAT_PATTERN.matcher(event.getCodedMessage()).matches()) {
+            openChatWhenContainerClosed = true;
```

Comment:
Should we have some kind of additional safeguards here? Like that the chat message should arrive with a container open? Or that it is reset whenever a container is opened? I'm just (slightly) worried that this bool is going to get in an incorrect state, somehow.

Reply:
Legacy impl is quite similar, I guess I could add a reset.

Reply:
It's up to you.